src/main/scala/spray/json/SnakeCaseJsonSupport.scala

```diff
@@ -0,0 +1,52 @@
+package spray.json
+
+import scala.reflect.ClassTag
+import scala.collection.mutable
+
+/**
+  * Provides snake cased JsonFormats
+  */
+trait SnakeCaseJsonSupport extends DefaultJsonProtocol {
+
+  override protected def extractFieldNames(classTag: ClassTag[_]): Array[String] = {
+    def snakify(name: String) = {
```

Comment:
Maybe we should also just use the regex here? Performance shouldn't be an issue as the formats are usually only evaluated once per type.